packages/davinci-client/src/lib/client.store.ts

```diff
@@ -96,7 +112,7 @@ export async function davinci<ActionType extends ActionTypes = ActionTypes>({
 
       const serverSlice = nodeSlice.selectors.selectServer(rootState);
 
-      return () => authorize(serverSlice, collector);
+      return () => authorize(serverSlice, collector, log);
```

Comment:
while for now this is non-blocking, i'd like us to consider how adding function parameters can affect composability later on.

obviously to achieve a fully composable API we will need to rethink some things.

Reply:
I agree that this is not optimal. I think we can reevaluate this when we introduce the new library we've been discussing. Until then, I'm recommending that we make this particular exception as these transformation functions contain important log data for debugging.